storage/diskmetricstore.go

```diff
@@ -125,7 +125,9 @@ func (dms *DiskMetricStore) Healthy() error {
 	// A pushgateway that cannot be written to should not be
 	// considered as healthy.
 	if len(dms.writeQueue) == cap(dms.writeQueue) {
-		return fmt.Errorf("write queue is full")
+		error_msg := "write queue is full"
+		level.Error(dms.logger).Log("msg", error_msg)
+		return fmt.Errorf(error_msg)
```

Comment:
In Go, local variables use camelCase. But I would just create the error directly. It will still log fine. Also, let's maybe use warn level, as you could legitimately be in an overloaded state and recover from it, and no harm was done.

In summary:

 ```suggestion
		err := fmt.Errorf("write queue is full")
		level.Error(dms.logger).Log("msg", err)
		return err
```